rust/lancedb/src/table.rs

```diff
@@ -1398,19 +1398,19 @@ impl Tags for NativeTags {
     }
 
     async fn create(&mut self, tag: &str, version: u64) -> Result<()> {
-        let dataset = self.dataset.get_mut().await?;
+        let dataset = self.dataset.get().await?;
```

Comment:
Tags don't mutate the table history itself, so we don't need to require that you have the latest table version checked out to create or modify tags.